src/features/weather/weatherSlice.js

```diff
@@ -0,0 +1,157 @@
+import axios from "axios"
+import moment from "moment"
+import { createSlice } from "@reduxjs/toolkit"
+
+export const weatherSlice = createSlice({
+    name: "counter",
+    initialState: {
+        currentCity: {}, //Current city displayed
+        previousCity: {}, //Last city searched
+        cities: [], //List of saved cities
+        error: "",
+    },
+    reducers: {
+        //Set current city to display
+        setCurrentCity: (state, action) => {
+            state.previousCity = state.currentCity || action.payload
+            state.currentCity = action.payload
+        },
+        //Add the city into the list on the top right
+        addCity: (state) => {
+            const findCity = state.cities.find((city) => city.id === state.currentCity.id)
+            if (!findCity) {
+                state.cities.push(state.currentCity)
+            }
+        },
+        //Set into current city the city that the user clicks
+        selectCity: (state, action) => {
+            const findCity = state.cities.find((city) => city.id === action.payload)
+            if (findCity) {
+                state.currentCity = findCity
+            }
+        },
+        //If error occurs and user clicks back then return to previous city
+        selectPreviousCity: (state) => {
+            state.currentCity = state.previousCity
+            state.error = ""
+        },
+        //Set the error in the state
+        setError: (state, action) => {
+            state.error = action.payload
+        },
+    },
+})
+
+export const { setCurrentCity, addCity, selectCity, selectPreviousCity, setError } = weatherSlice.actions
+
+//Get the correct name in base of speed (m/s)
+const getTypeOfWind = (speed) => {
+    if (speed < 3.5) return `Light wind`
+    if (speed < 8) return `Moderate wind`
+    if (speed < 24) return `Strong wind`
+    if (speed < 30) return `Storm`
```

Comment:
Where these values come from? (3.5, 8., 24, 30) 
Maybe a few comments would help here

Reply:
Yes you are right.
This values comes to web, in the api I haven't this information to set in the description in the month tab, so I do some research and I find come range to define better the wind speed

Reply:
I see,  so maybe we can define an object of values so we can easily change them?

I also believe that we can find a better strategy instead of a sequence of `if`

For example we can map an array like this:

```
[{minSpeed: 0, maxSpeed: 3.5, value: 'Light Wind'}]
```

and then do a search where the `speed` value is between max and min values

What do you think?